pfSense-pkg-API/files/etc/inc/api/models/APISystemCRLCreate.inc

```diff
@@ -0,0 +1,155 @@
+<?php
+//   Copyright 2022 Jared Hendrickson
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+require_once("api/framework/APIModel.inc");
+require_once("api/framework/APIResponse.inc");
+
+
+class APISystemCRLCreate extends APIModel {
+    public $cert_method;
+
+    # Create our method constructor
+    public function __construct() {
+        parent::__construct();
+        $this->privileges = ["page-all", "page-system-certmanager"];
+        $this->change_note = "Added certificate revocation list via API";
+        $this->cert_method = null;
+    }
+
+    public function action() {
+        # Add the new certificate
+        $this->config["crl"][] = $this->validated_data;
+
+        # Write changes to configuration
+        $this->write_config();
```

Comment:
I'm not sure calling for OpenVPN and IPSec restart is necessarily since this is newly created CRL and not yet been utilized by either one. Adding this functionality will create unnecessary overhead.
At least from what i can see OpenVPN can utilize CRL created from API without need of restart.

Reply:
Since the `cert` field is not present, you're correct. This will be fine for now.